Doc/whatsnew/3.13.rst

```diff
@@ -2187,6 +2214,24 @@ Changes in the Python API
   Calling it with argument ``2`` restores the current default behavior.
   (Contributed by Serhiy Storchaka in :gh:`66410`.)
 
+* Calling :func:`locals` in an :term:`optimised scope` now produces an
+  independent snapshot on each call, and hence no longer implicitly updates
+  previously returned references. Obtaining the legacy CPython behaviour now
+  requires explicit calls to update the initially returned dictionary with the
+  results of subsequent calls to ``locals()``. (Changed as part of :pep:`667`.)
+
+* Calling :func:`locals` from a comprehension at module or class scope
+  (including via ``exec`` or ``eval``) once more behaves as if the comprehension
+  were running as an independent nested function (i.e. the local variables from
+  the containing scope are not included). In Python 3.12, this had changed
+  to include the local variables from the containing scope when implementing
+  :pep:`709`. (Changed as part of :pep:`667`.)
+
```

Comment:
@ncoghlan would this be an appropriate place for `..versionchanged :: python3.12` ?

Reply:
There are notes on PEP 709 in the Python 3.12 What's New. I only mentioned the Python 3.12 change here because this particular aspect of the PEP 667 changes is actually reverting the `locals()` behaviour back to the way it behaved in Python 3.11.